### PR TITLE
refine runClients func for mixer performance

### PR DESCRIPTION
**Pull request by lichuqiang**
refine runClients func for mixer performance

### CODE REVIEW

File: mixer/pkg/perf/controller.go [hunk 1]
```diff
@@ -92,18 +99,46 @@ func (c *Controller) initializeClients(address string, setup *Setup) error {
...
+			resCount++
+			if e != nil {
+				resErr = append(resErr, e)
+			}
+			if resCount == len(c.clients) {
+				// All of the calls returned
+				if len(resErr) > 0 {
+					err = errorsutil.AggregateError(resErr)
+				}
+				return
```
Comment (hzxuzhonghu) [inline mixer/pkg/perf/controller.go]: should return err

Comment (lichuqiang) [reply mixer/pkg/perf/controller.go]: updated

**Comment by lichuqiang** [comment]
@geeknoid @costinm PTAL :)

**Comment by lichuqiang** [comment]
PR in active state

File: mixer/pkg/perf/controller.go [hunk 2]
```diff
@@ -92,18 +99,47 @@ func (c *Controller) initializeClients(address string, setup *Setup) error {
...
+				resErr = append(resErr, e)
+			}
+		case <-timer.C:
+			return fmt.Errorf("timeout waiting for call response")
 		}
 	}
 
-	return err
+	if len(resErr) > 0 {
+		return errorsutil.AggregateError(resErr)
```
Comment (ozevren) [inline mixer/pkg/perf/controller.go]: Most of the Istio code is using multierror.Append to accumulate errors.  We should standardize on that.

File: mixer/pkg/perf/controller.go [hunk 3]
```diff
@@ -15,13 +15,20 @@
...
 	"net/rpc"
+	"time"
+
+	errorsutil "github.com/onsi/gomega/gstruct/errors"
 
 	"istio.io/istio/pkg/log"
 )
 
+// TODO(lichuqiang): modify defaultTimeout accordingly.
+const defaultTimeout = time.Duration(30 * time.Second)
```
Comment (ozevren) [inline mixer/pkg/perf/controller.go]: This is likely to be too restrictive. When running perf tests, I've used 1 or 5 minute intervals to get some stable results.

**Comment by lichuqiang** [comment]
@ozevren comments addressed. PTAL.

**Pull request merged**